src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,33 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Employee[] employees = new Employee[names.length];
+
+        for (int i = 0; i < names.length; i++) {
+            employees[i] = new Employee(names[i]);
+        }
```

Comment:
Don’t create redundant loops. Two is enough.
The more loops you have, the slower your code may get, so try to solve task with minimum amount.
Now I see 4 loops in the solution, think about how to refactor it, and improve the performance of the solution.